src/Node.java

```diff
@@ -20,6 +20,9 @@
     public Node(Type nodeType, int inputCount) {
         this.nodeType = nodeType;
 
+        weights = new ArrayList<>();
```

Comment:
Initialized with declaration, this is not necessary

Reply:
thats gross